src/map/achievement.c

```diff
@@ -0,0 +1,1265 @@
+		else {
+			entry->mapindex = map_mapname2mapid(mapname);
+
+			if (entry->mapindex == -1)
+				ShowWarning("achievement_read_db_sub: Invalid map name %s for achievement %d in \"%s\".\n", mapname, achievement_id, source);
+		}
+	}
+
+	if ((t = config_setting_get_member(cs, "dependent"))) {
+		if (config_setting_is_aggregate(t)) {
+			int i, len = config_setting_length(t);
+
+			for (i = 0; i < len && entry->dependent_count < MAX_ACHIEVEMENT_DEPENDENTS; i++) {
+				RECREATE(entry->dependents, struct achievement_dependent, entry->dependent_count + 1);
+				entry->dependents[entry->dependent_count].achievement_id = config_setting_get_int_elem(t, i);
+				entry->dependent_count++;
+			}
+		} else
+			ShowWarning("achievement_read_db_sub: Invalid dependent format for achievement %d in \"%s\".\n", achievement_id, source);
+	}
+
+	if ((t = config_setting_get_member(cs, "reward"))) {
+		if (config_setting_is_group(t)) {
+			if (config_setting_get_member(t, "itemid")) {
+				int nameid = 0;
+
+				if (config_setting_lookup_int(t, "itemid", &nameid) && itemdb_exists(nameid)) {
+					entry->rewards.nameid = nameid;
+					entry->rewards.amount = 1; // Default the amount to 1
+				} else if (nameid && !itemdb_exists(nameid)) {
+					ShowWarning("achievement_read_db_sub: Invalid reward item ID %hu for achievement %d in \"%s\". Setting to 0.\n", nameid, achievement_id, source);
+					entry->rewards.nameid = nameid = 0;
+				}
+
+				if (config_setting_get_member(t, "amount")) {
+					int amount = 0;
+
+					if (nameid && config_setting_lookup_int(t, "amount", &amount) && amount > 0)
+						entry->rewards.amount = amount;
+				}
+			}
+			if (config_setting_get_member(t, "script")) {
+				const char *script_char = NULL;
+
+				if (config_setting_lookup_string(t, "script", &script_char))
+					entry->rewards.script = parse_script(script_char, source, achievement_id, SCRIPT_IGNORE_EXTERNAL_BRACKETS);
+			}
+			if (config_setting_get_member(t, "titleid")) {
+				int title_id = 0;
+
+				if (config_setting_lookup_int(t, "titleid", &title_id) && title_id > 0)
+					entry->rewards.title_id = title_id;
+			}
+		}
+	}
+
+	if (config_setting_lookup_int(cs, "score", &score) && score > 0)
+		entry->score = score;
+
+	return entry;
+}
+
+/**
+ * Loads achievements from the achievement db.
+ */
+void achievement_read_db(void)
+{
+	struct config_setting_t *adb = NULL, *a = NULL;
+	int i = 0;
+	const char *dbsubpath[] = {
+		"",
+		"/"DBIMPORT"/",
+		//add other path here
+	};
+
+	for (i = 0; i < ARRAYLENGTH(dbsubpath); i++) {
+		char filepath[256];
+		int count = 0;
+
+		if (!i)
+			sprintf(filepath, "%s/%s%s%s", db_path, DBPATH, dbsubpath[i], "achievement_db.conf");
+		else
+			sprintf(filepath, "%s%s%s", db_path, dbsubpath[i], "achievement_db.conf");
+
+		if (config_read_file(&achievement_db_conf, filepath))
```

Comment:
please use `conf_read_file` in common/conf.h instead of `config_read_file` in libconfig.h.

if still want to use `config_read_file`, then this line need change to:

```c
if (config_read_file(&achievement_db_conf, filepath) != CONFIG_TRUE)
```